# Profitable App Profiles for the App Store and Google Play Markets

In this project we are looking through the App Store and Google Play Market, to find the profitable apps.  The data for App store can be found [here](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps) and for the Google Play Market [here](https://www.kaggle.com/datasets/lava18/google-play-store-apps) From this we can find the apps that are the most profitable.  This can help people who are developing apps to see what attracts the most people.


# Exploring the Data

We take a look at what the two datasets and see how we can relate the two datasets together

In [1]:
# Opening AppleStore.csv and googleplaystore.csv
# Storing the data in variables

from csv import reader

# Opening files
open_applestore = open('AppleStore.csv')
open_googleplay = open('googleplaystore.csv')

# Reading files
read_applestore = reader(open_applestore)
read_googleplay = reader(open_googleplay)

# Putting data into list from files
apple_data = list(read_applestore)
google_data = list(read_googleplay)

# Creating header from data set
apple_header = apple_data[0]
google_header = google_data[0]

# Removing header from data set
apple_data = apple_data[1:]
google_data = google_data[1:]

In [2]:
# Looking through data
def explore_data(data_set, start, end, row_columns = False):
    for row in data_set[start:end]:
        print(row)
        print('\n')
    if row_columns:
        print('Number of rows', len(data_set))
        print('Number of columns', len(data_set[0]))
        
print(apple_header)
print('\n')
explore_data(apple_data, 0, 5, True)
        

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows 7197
Number of columns 16


The apple data has 7197 rows and 16 columns

In [3]:
print(google_header)
print('\n')
explore_data(google_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

The google data has 10841 rows and 13 columns

# Data Cleaning

In the discussion section for the Google Play Dataset they talk about how row 10472 has an error, we can start by checking this out and deleting it if it's true.

In [4]:
print(google_data[10472])
print('/n')
print(google_header)
print('/n')
print(google_data[0])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
/n
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
/n
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


We can see that row 10472 is missing Category data so we can go ahead and delete the row from the dataset

In [5]:
print('Length before:', len(google_data))
del(google_data[10472])        
print('Length after:', len(google_data))

Length before: 10841
Length after: 10840


We can see that the row has been removed by looking at the length of the dataset before and after

## Removing Duplicates
Now we take a look to see if there are any duplicate data in the dataset.

In [6]:
duplicates = []
unique = []

for element in google_data:
    name = element[0]
    if name in unique:
        duplicates.append(name)
    else:
        unique.append(name)

print('Number of duplicates', len(duplicates))

    

Number of duplicates 1181


We can see that there are 1181 duplicate apps from the Google Play Dataset.  Now we taking a look at the duplicates and see if there are any difference between them. 

In [7]:
print(duplicates[:10])

['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In [8]:
for x in google_data:
    name = x[0]
    if name == 'Slack':
        print(x)

['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']


After looking at one of the duplicate apps we notice that number of reviews for one of them is different from the other two.  When choosing which one too keep we will take the one with the most reviews.

In [9]:
reviews_max = {}

for app in google_data:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
print('Expected length:', len(google_data) - 1181)
print('Actual length:', len(reviews_max))
        

Expected length: 9659
Actual length: 9659


We can see that the number of expected duplicates match with the number of actual. We subtracted the entire length of the google data set by the number of duplicates we got before which was 1181 and got 9659.  Now we clean up the data by removing the duplicates

In [10]:
google_clean = []
already_added = []

for app in google_data:
    name = app[0]
    n_reviews = float(app[3])
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        google_clean.append(app)
        already_added.append(name)  


In [11]:
explore_data(google_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows 9659
Number of columns 13


## Removing Non-English Apps

Next we work on removing apps in different languages since we only want to work with english apps for easy use.  We can see an example of a non-english app below and we want to get rid of these

In [12]:
print(apple_data[813][1])
print(apple_data[6731][1])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


We create a function that can determine if an app is in english or not. The function looks at each character from the name of the app and if it has more then three non-english characters then it is determine to be non-english.

In [13]:
def english(string):
    count = 0
    for character in string:
        if ord(character) > 127:
            count += 1
    if count > 3:
        return False
    else:
        return True
print(english('Instagram'))
print(english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english('Docs To Go™ Free Office Suite'))
print(english('Instachat 😜'))

True
False
True
True


Cleaning the Google Play and Apple Datasets by removing non-english apps using the english function

In [14]:
english_google = []
english_apple = []
non_english_app = []
for app in google_clean:
    name = app[0]
    if english(name):
        english_google.append(app)

for app in apple_data:
    name = app[1]
    if english(name):
        english_apple.append(app)

explore_data(english_google, 0, 3, True)
print('\n')
explore_data(english_apple, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows 9614
Number of columns 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'G

## Finding the Free Apps

The free apps are then filtered into a new list to use for our analysis

In [15]:
google_final = []
apple_final = []

for app in english_google:
    price = app[7]
    if price == '0':
        google_final.append(app)
        
for app in english_apple:
    price = app[4]
    if price == '0.0':
        apple_final.append(app)
        
print(len(google_final))
print(len(apple_final))

8864
3222


Now we have our final list for both Google and Apple.  We started with 10841 rows for Google and 7197 for Apple.  Now we are at 8864 for Google and 3222 for Apple.  We cleaned up the dataset quite a bit to start analyzing the data we are finally at the starting point.

## Analyzing the Data

We want to see what free apps perform the best in the form of frequency tables. This shows us what the most popular apps have in common. To do this we need to create functions to help display the results.

In [16]:
def freq_table(dataset, index):
    table ={}
    total = 0
    for app in dataset:
        total += 1
        element = app[index]
        if element in table:
            table[element] += 1
        else:
            table[element] = 1
            
    table_percentages = {}
    for key in table:
            percentage = (table[key] / total) * 100
            table_percentages[key] = percentage
    return table_percentages
  
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
        

We can take a look at the headers to see which columns we believe will give us a good insight.  A good place to start will be Genres.

In [17]:
google_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [18]:
apple_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

Now we use the display table function to help show which genres are the most popular in the App Store and Google Play store

In [19]:
display_table(apple_final, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


From the Apple dataset we can see that Games as the primary genre have the most attraction by far claiming over 50% of the free app market space.  Its hard to look more into seeing which games are more successful then others due to data not giving a sub category to help filter it even more.

In [20]:
display_table(google_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

From the Google Play Dataset we can see that Family, Game, and Tools are the top three categorys.  Now we can see if we get a better idea of what these three categories mean.  What is a family app? What is a tool app? What kind of games are played more?  If we take a look at the genre column this could help give us a better idea.

In [21]:
display_table(google_final, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

From looking at the genre table its hard to tell what is considered family.  Its most likely due to the age requirement for an app in the Google Play Store.  For Tools are the same across category and genre.  For games we can take a guess if the top genre is Entertainment, Education or Sports.  Also Games have multiple genres just by looking at the frequency table.

Now we take a look the number of installs for each genre.

In [22]:
genres_apple = freq_table(apple_final, 11)

for genre in genres_apple:
    total = 0
    len_genre = 0
    for app in apple_final:
        genre_app = app[11]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


For the App Store dataset it doesn't have an install column like the Google Play dataset, so to find the number of installs another way.  We added the total of rating_count_tot for each genre and divided that by the total number of apps in that genre. 

We can see that Navigation, Reference and Social Networking have the highest numbers

In [23]:
for x in apple_final:
    if x[11] == 'Navigation':
        print(x[1],":",x[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [24]:
for x in apple_final:
    if x[11] == 'Reference':
        print(x[1],":",x[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [25]:
for x in apple_final:
    if x[11] == 'Social Networking':
        print(x[1],":",x[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

We can see that navigation has 6 apps and the apps with the higher values have a lot more weight when getting the average.  The same can be said for Reference and Social Networking.  So it seems like these three genres popularity are carried by a few apps.  This is something to keep in mind when someone is trying develop an app for a certain genre.

Now we can take a look at the Google Play installs for each category.

In [26]:
category_google = freq_table(google_final,1)

for x in category_google:
    total = 0
    len_category = 0
    for y in google_final:
        category_app = y[1]
        if category_app == x:
            installs = y[5]
            installs = installs.replace('+','')
            installs = installs.replace(',','')
            installs = float(installs)
            total += installs
            len_category += 1
    average = total/len_category
    print(x,':',average)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

It looks like Communication, Social and Video Players are the most installed apps from the Google Play Store.  This is based on finding the average from all installed apps based on category.

## Conclusion

From this analysis of the two datasets, we started by cleaned the data.  First we removed the duplicates, then remove the non-english apps and finally remove the non-free apps.  Then we analyzed the data to see which apps are most profitable.  We started by looking at genre for each dataset.  For App store the most popular was Games and Google Play Store it was Family, Games and Tools.  Next we look at the number of apps installed by genre and we can see that for both datasets Social apps rank among the top three.  For someone who wants to make a profitable app for both the App store and Google Play Store either an app built for Games or Social.  The biggest obstacle is the amount of competition they will have in each space as many people build apps in these genres.  Depending on the person's knowledge you can 